codebase/entity_connexion.py

```diff
@@ -1,83 +1,143 @@
 
     def get_transactions(self):
+        """Get list of transaction involving each Account assigned to the Entity."""
+
         transaction_tables = []
         for account in self.accounts:
             transaction_tables.append(account.get_transactions())
 
         if len(transaction_tables) == 1 and transaction_tables[0] is None:
             return
+
         transactions = pd.concat(transaction_tables)
         transactions['datetime'] = pd.to_datetime(transactions.index)
 
         column_list = ['datetime', 'amount',
                        'transferringAccount_id', 'transferringAccountName', 'transferringAccountType',
                        'acquiringAccount_id', 'acquiringAccountName', 'acquiringAccountType',
                        'transactionTypeMain', 'transactionTypeSupplementary', 'unitType']
-        for c in column_list:
-            if c not in transactions.columns:
-                warnings.warn('Columns not available. Investigate!')
-                return
+
+        if not set(column_list).issubset(transactions.columns):
+            warnings.warn('At least one of the required columns is missing from the transaction table. Investigate!')
+            return
 
         transactions = transactions[column_list].copy()
 
         if self.entity_type == 'Account':
-            for v in ['transferring', 'acquiring']:
-                transactions.rename(columns={f'{v}Account_id': f'{v}Entity_id',
-                                             f'{v}AccountName': f'{v}Entity_name',
-                                             f'{v}AccountType': f'{v}Entity_type'}, inplace=True)
+            # rename AccountHolder -> Entity
+            for side in ['transferring', 'acquiring']:
+                transactions.rename(columns={f'{side}Account_id': f'{side}Entity_id',
+                                             f'{side}AccountName': f'{side}Entity_name',
+                                             f'{side}AccountType': f'{side}Entity_type'}, inplace=True)
         elif self.entity_type == 'AccountHolder':
-            # todo: merge with account get accountholder_id > merger with accountHolder (transferring and acquiring) > keep accountholder name
-            #  then rename columns, drop transactions from accountholder_id to itself
-            pass
+            # map each (transferring or acquiring) account to its account holder
+            for side in self.sides:
+                account_to_accountHolder_temp = account_to_accountHolder.copy()
+                account_to_accountHolder_temp.rename(columns={"account_id": f"{side}Account_id",
+                                                              "accountHolder_id": f"{side}AccountHolder_id",
+                                                              "accountHolder_name": f"{side}AccountHolder_name"}, inplace=True)
+                transactions = transactions.merge(account_to_accountHolder_temp)
+
```

Comment:
I would always add a test that all has been merged well:

`merge(...., indicator=True)`

Also only merge left:

`merge(..., how='left')

Then check:

`merge_test = df.groupby('_merge', asindex=False)['_merge'].count()`
`if len(merge_test[merge_test._merge != 'both'])>1 ValueError`

Reply:
ok, check implémenté